src/sentry/rules/conditions/event_frequency.py

```diff
@@ -281,14 +290,17 @@ def get_rate(
         environment_id: int,
         comparison_type: str,
     ) -> int:
-        start, end = self.get_comparison_start_end(timedelta(), duration)
+        current_time = timezone.now()
+        start, end = self.get_comparison_start_end(current_time, duration)
         with self.disable_consistent_snuba_mode(duration):
             result = self.query(event, start, end, environment_id=environment_id)
             if comparison_type == ComparisonType.PERCENT:
                 # TODO: Figure out if there's a way we can do this less frequently. All queries are
                 # automatically cached for 10s. We could consider trying to cache this and the main
                 # query for 20s to reduce the load.
-                start, end = self.get_comparison_start_end(comparison_interval, duration)
+                start, end = self.get_comparison_start_end(
```

Comment:
RE: comments above

1 - lets make this kwargs
2 - lets pre-calucate the `end` here

Reply:
+1 on just subtracting the `comparison_interval` from the `end` here since we are already doing the logic to check for comparisons here.

Perhaps we rename the function to like `get_query_window` or something a little more descriptive to it as well.